hack/generator/pkg/astmodel/type_name.go

```diff
@@ -52,9 +52,12 @@ func (typeName *TypeName) AsType(codeGenerationContext *CodeGenerationContext) a
 	return ast.NewIdent(typeName.name)
 }
 
-// References indicates whether this Type includes any direct references to the given Type
-func (typeName *TypeName) References(d *TypeName) bool {
-	return typeName.Equals(d)
+// References returns a set containing this type name.
+func (typeName *TypeName) References() TypeNameSet {
```

Comment:
A convention I like for these kinds of methods is to prefix with `As`  or `To` (usually _As_ 😁 )

```suggestion
func (typeName *TypeName) References() AsTypeNameSet {
```

Reply:
Confused on this suggestion @theunrepentantgeek - are you suggesting to change the name of `TypeNameSet` to `AsTypeNameSet`? That seems weird to me.

Did you maybe mean `AsReferences()` or something? (Note I am not sure I love that either but I think it makes more sense than suggested change diff I'm seeing above)

Reply:
I did say I was tired. 😁 

I must have reverted to C style "type name" thinking ... 'cause I though the method was named `TypeNameSet` and I though `AsTypeNameSet` would be a better name. #sigh